Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/DaysSinceOnsetOfSymptomsVectorDeterminator.kt

```diff
@@ -0,0 +1,43 @@
+package de.rki.coronawarnapp.submission
+
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.numberOfDayChanges
+
+class DaysSinceOnsetOfSymptomsVectorDeterminator {
+
+    companion object {
+        const val VECTOR_LENGTH = 15
+    }
+
+    @Suppress("MagicNumber")
+    internal fun determine(symptoms: Symptoms): DaysSinceOnsetOfSymptomsVector {
+        return when (symptoms.symptomIndication) {
+            Symptoms.Indication.POSITIVE -> {
+                when (symptoms.startOfSymptoms) {
```

Comment:
I would put that second when Block in a seperate method for better readability. But it's just a personal preference...